File: msgQ_client.py
import urllib2
import httplib

class msgclient:
    def __init__(self, hostname, port):
        qServerUrl = "http://"+hostname + ":" + str(port) + "/"
        self._url=qServerUrl
        self._hostname=hostname
        self._port = str(port)
        
    def get(self, qName):
        url = self._url+qName
        
        try:
            data = urllib2.urlopen(url).read()
        except urllib2.HTTPError, e:
            print "HTTP error: %d" % e.code
        except urllib2.URLError, e:
            print "Network error: %s" % e.reason.args[1]
        return data

    def post(self, qName, value):
        url = self._url+qName
        try:
            data = urllib2.urlopen(url, value).read()
        except urllib2.HTTPError, e:
            if e.code==201: #Not really an error as that is the expected response
                data = e.msg
            else:
                print "HTTP error: %d" % e.code
                print repr(dir(e))
                print e.msg
        except urllib2.URLError, e:
            print "Network error: %s" % e.reason.args[1]
            raise
        return data

    def put(self, qName):
        conn = httplib.HTTPConnection(self._hostname + ":" + self._port)
        conn.request("PUT", "/" + qName)
        r1 = conn.getresponse()
        print r1.status, r1.reason
        data1 = r1.read()
        conn.close()

File: msgQ_client.php
<?php

require_once "RESTclient.php";

class msgclient {
  public $baseUrl;// = "http://localhost:8990/";

  function __construct($b) {
    print "Construct ".$b."\n";
    $this->baseUrl = $b;
  }

  function get($qName){
    $rest = new RESTclient();
    
    $url = $this->baseUrl."msg/".$qName;
    $rest->createRequest("$url","GET");
    $rest->sendRequest();
    $output = $rest->getResponse();
    return $output;
  }
  
  function put($qName){
    $rest = new RESTclient();
    
    $url = $this->baseUrl.$qName;
    //$url = "http://localhost:8990/".$qName;
    $rest->createRequest("$url","PUT");
    $rest->sendRequest();
    $output = $rest->getResponse();
    return $output;
  }
  
  
  function post($qName, $value){
    $rest = new RESTclient();
    
    //$inputs = array();
    //$inputs["appid"] = "YahooDemo";
    
    $url = "http://localhost:8990/".$qName;
    //$rest->createRequest("$url","POST", $inputs);
    $rest->createRequest("$url","POST", $value);
    $rest->sendRequest();
    $output = $rest->getResponse();
    return $output;
  }
}
?>
